sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileTable.scala

```diff
@@ -77,11 +77,19 @@ abstract class FileTable(
     }
   }
 
+  lazy val dataSchema: StructType = {
+    StructType(fileSchema.filterNot(partitionSchema.contains))
```

Comment:
should we consider `caseSensitive` option as line 106 ~ line 109:
```
val fields = fileSchema.fields.filterNot { field =>
    val colName = PartitioningUtils.getColName(field, caseSensitive)
    partitionNameSet.contains(colName)
}

```

Reply:
Yes. Let's see if there are other failures.

Reply:
3 test in `HiveThriftHttpServerSuite ` failed, seems these three use cases fail in other pr too ....